pkg/envtest/binaries.go

```diff
@@ -124,15 +142,22 @@ func downloadBinaryAssets(ctx context.Context, binaryAssetsDirectory, binaryAsse
 		}
 	}
 
+	exact, major, minor := interpretKubernetesVersion(binaryAssetsVersion)
+
 	var binaryAssetsIndex *index
-	if binaryAssetsVersion == "" {
+	if binaryAssetsVersion != "" && exact != "" {
+		// Look for these specific binaries locally before downloading them from the release index.
+		// Use the canonical form of the version from here on.
+		binaryAssetsVersion = "v" + exact
+	} else if binaryAssetsVersion == "" || major != 0 || minor != 0 {
+		// Select a stable version from the release index before continuing.
 		var err error
 		binaryAssetsIndex, err = getIndex(ctx, binaryAssetsIndexURL)
 		if err != nil {
 			return "", "", "", err
 		}
```

Comment:
Should we add an else branch here that returns to cover error cases?

Reply:
Done.